backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/StaticObjects.kt

```diff
@@ -86,22 +84,22 @@ internal fun StaticData.createInitializer(type: IrClass, vararg fields: ConstVal
 internal fun StaticData.createConstArrayList(array: ConstPointer, length: Int): ConstPointer {
     val arrayListClass = context.ir.symbols.arrayList.owner
 
-    val arrayListFqName = arrayListClass.fqNameForIrSerialization
     val arrayListFields = mapOf(
-        "$arrayListFqName.array" to array,
-        "$arrayListFqName.offset" to Int32(0),
-        "$arrayListFqName.length" to Int32(length),
-        "$arrayListFqName.backing" to NullPointer(kObjHeader))
+        "array" to array,
+        "offset" to Int32(0),
+        "length" to Int32(length),
+        "backing" to NullPointer(kObjHeader))
 
     // Now sort these values according to the order of fields returned by getFields()
     // to match the sorting order of the real ArrayList().
-    val sorted = linkedMapOf<String, ConstValue>()
+    val sorted = mutableListOf<ConstValue>()
     context.getLayoutBuilder(arrayListClass).fields.forEach {
-        val fqName = it.fqNameForIrSerialization.asString()
-        sorted.put(fqName, arrayListFields[fqName]!!)
+        if (it.parent == arrayListClass) {
```

Comment:
Maybe make it a `require`? Now it seems to just skip fields from superclasses (if any), so this code would silently break if a superclass actually gets a field, leading to later vague LLVM failure.